### PR TITLE
Improve the Redux docs

### PR DESCRIPTION
**Pull request by cooperka**
Things done in this PR:

1. Update the ReduxExample app to be more robust, using `getStateForAction` and checking for `null` states
2. Document the use of `getStateForAction` with redux
3. Link to the ReduxExample app in the docs
4. Clean up the Contributors guide, and make sure there's only one source of truth for how to run the examples

Among other things, this should help clarify some of the confusions expressed in https://github.com/react-community/react-navigation/issues/485.

### CODE REVIEW

**Comment by grabbou** [comment]
Can we get this rebased on top of master (remove `LinkingExample`)?

**Comment by cooperka** [comment]
@grabbou done.

File: examples/ReduxExample/src/reducers/index.js [hunk 1]
```diff
@@ -4,27 +4,31 @@ import { NavigationActions } from 'react-navigation';
...
-    { key: 'InitA', routeName: 'Main' },
-    { key: 'InitB', routeName: 'Login' },
-  ],
-};
-
-const initialAuthState = { isLoggedIn: false };
+const firstAction = AppNavigator.router.getActionForPathAndParams('Main');
+const tempNavState = AppNavigator.router.getStateForAction(firstAction);
+const secondAction = AppNavigator.router.getActionForPathAndParams('Login');
+const initialNavState = AppNavigator.router.getStateForAction(secondAction, tempNavState);
```
Comment (grabbou) [inline examples/ReduxExample/src/reducers/index.js]: I get the point, tho I believe this will be more complex for new-commers than the plain object.

Comment (cooperka) [inline examples/ReduxExample/src/reducers/index.js]: Unfortunately, I agree it's more complicated, but it does also prevent silly user errors. Given the number of people who couldn't figure out how to use it and got errors like in #485 it seems like this is a better approach. I'd love to write it a simpler way if possible -- any ideas?

Comment (cooperka) [inline examples/ReduxExample/src/reducers/index.js]: I could simply leave the initial state there as comments, to clarify what this logic all turns into. Or I could add a helper method that generates a state based on an array of actions --
 e.g. `AppNavigator.router.getInitialStateForPath('Main')` which could also take an array `AppNavigator.router.getInitialStateForPath(['Main', 'Login'])`. Seems like a very common pattern that could warrant a helper method like this.

Comment (cooperka) [inline examples/ReduxExample/src/reducers/index.js]: Though perhaps they could go in a future PR so we get this in sooner? Imo this is better than the current state.

Comment (grabbou) [inline examples/ReduxExample/src/reducers/index.js]: All righty, I'll leave that up to you since I don't use `Redux` with `react-navigation` actively.

Comment (cooperka) [inline examples/ReduxExample/src/reducers/index.js]: Ok sounds good, I'll see what I can do.

**Pull request merged**